### PR TITLE
Add map restart and map change message

### PR DESCRIPTION
**Pull request by wrefgtzweve**
Picked a random emoji and the text might not be the best, feel free to commit change those im not great at picking such stuff. Other than that, it should work i think it's not that complicated.

Might be fancy to add the join url to it, but that might cause issues if we have multiple servers running relay and only one join url

I've tested the systime method in p2p and it works, i assumed it worked on dedicated servers too.

### CODE REVIEW

File: moon/cfc_chat_transit/server/modules/startup.moon [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    if SysTime() > 500 -- If systime is over 500, the server switched maps. Else it's a hard restart
+        eventText = "Map switched to " .. map
+    else
+        eventText = "Server restarted! Current map is " .. map
+
+    @Send
+        Type: "map_init"
+        Data:
+            eventText: GetMap!
```
Comment (legokidlogan) [inline moon/cfc_chat_transit/server/modules/startup.moon]:
```suggestion
            Content: eventText
```

File: web/discord_relay/queue.go [hunk 2]
```diff
@@ -41,6 +41,7 @@ const (
 	EMOJI_HALTED  = "<:halted:398133588010336259>"
 	EMOJI_BUILD   = "<:build:933512140395012107>"
 	EMOJI_PVP     = "<:bk:812130062379515906>"
+	EMOJI_MAP     = "📊"
```
Comment (legokidlogan) [inline web/discord_relay/queue.go]:
```suggestion
	EMOJI_MAP     = "🗺️"
```

**Review by legokidlogan — CHANGES_REQUESTED** [review]

**Comment by legokidlogan** [comment]
I'd also personally recommend giving the map event its own unique color to help it stand out from the other messages, especially since there will be a flurry of green connection messages surrounding it on changelevels. Maybe something like `COLOR_TEAL = 0x14CEC2`

File: moon/cfc_chat_transit/server/modules/startup.moon [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import guard from ChatTransit
+import GetMap from game
+
+ChatTransit.Initialize = (data) =>
```
Comment (sarahsturgeon) [inline moon/cfc_chat_transit/server/modules/startup.moon]: `Initialize` has some code connotations, I think it'd be better to name it a little more bluntly
```suggestion
ChatTransit.MapStartup = (data) =>
```

**Review by sarahsturgeon — DISMISSED** [review]

File: moon/cfc_chat_transit/server/modules/startup.moon (hunk 3 above)
Comment (legokidlogan) [reply moon/cfc_chat_transit/server/modules/startup.moon]: Along with `hook.Add "Initialize", "CFC_ChatTransit_StartupListener", guard ChatTransit\MapStartup` of course

File: web/discord_relay/queue.go (hunk 2 above)
Comment (wrefgtzweve) [reply web/discord_relay/queue.go]: Up to phatso

Comment (overlordbleck) [reply web/discord_relay/queue.go]: I like the map :)

**Review by overlordbleck — APPROVED** [review]

File: web/discord_relay/queue.go (hunk 2 above)
Comment (wrefgtzweve) [reply web/discord_relay/queue.go]: he likes the map

**Review by legokidlogan — APPROVED** [review]

File: moon/cfc_chat_transit/server/modules/startup.moon [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import guard from ChatTransit
+import GetMap from game
+
+ChatTransit.MapStartup = (data) =>
+    eventText = ""
+    map = GetMap!
+
+    if SysTime() > 500 -- If systime is over 500, the server switched maps. Else it's a hard restart
```
Comment (plally) [inline moon/cfc_chat_transit/server/modules/startup.moon]:
```suggestion
    if SysTime! > 500 -- If systime is over 500, the server switched maps. Else it's a hard restart

```

**Review by sarahsturgeon — DISMISSED** [review]
valid

**Pull request merged**